esti/s3_gateway_test.go

```diff
@@ -289,6 +289,86 @@ func TestS3IfNoneMatch(t *testing.T) {
 	}
 }
 
+func TestListMultipartUploads(t *testing.T) {
+	blockStoreType := viper.GetString(ViperBlockstoreType)
+	if blockStoreType != "s3" {
+		t.Skip("Skipping test - blockstore type is not s3")
+	}
+	ctx, logger, repo := setupTest(t)
+	defer tearDownTest(repo)
+	s3Endpoint := viper.GetString("s3_endpoint")
+	s3Client := createS3Client(s3Endpoint, t)
+	multipartNumberOfParts := 3
+	multipartPartSize := 5 * 1024 * 1024
+
+	// create two objects for two mpus
+	obj1 := "object1"
+	obj2 := "object2"
+	path1 := "main/" + obj1
+	path2 := "main/" + obj2
+
+	input1 := &s3.CreateMultipartUploadInput{
+		Bucket: aws.String(repo),
+		Key:    aws.String(path1),
+	}
+	input2 := &s3.CreateMultipartUploadInput{
+		Bucket: aws.String(repo),
+		Key:    aws.String(path2),
+	}
+	// create first mpu
+	resp1, err := s3Client.CreateMultipartUpload(ctx, input1)
+	require.NoError(t, err, "failed to create multipart upload")
+	parts := make([][]byte, multipartNumberOfParts)
+	for i := 0; i < multipartNumberOfParts; i++ {
+		parts[i] = randstr.Bytes(multipartPartSize + i)
+	}
+
+	completedParts1 := uploadMultipartParts(t, ctx, s3Client, logger, resp1, parts, 0)
+
+	completeInput1 := &s3.CompleteMultipartUploadInput{
+		Bucket:   resp1.Bucket,
+		Key:      resp1.Key,
+		UploadId: resp1.UploadId,
+		MultipartUpload: &types.CompletedMultipartUpload{
+			Parts: completedParts1,
+		},
+	}
+	// check first mpu appears
+	output, err := s3Client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{Bucket: resp1.Bucket})
+	require.NoError(t, err, "error listing multiparts")
+	str := extractUploadKeys(output)
+	require.Contains(t, str, obj1)
+
+	// create second mpu check both appear
+	_, err = s3Client.CreateMultipartUpload(ctx, input2)
+	require.NoError(t, err, "failed to create multipart upload")
+	output, err = s3Client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{Bucket: resp1.Bucket})
+	str = extractUploadKeys(output)
+	require.Contains(t, str, obj1)
+	require.Contains(t, str, obj2)
+
+	// finish first mpu check only second appear
+	_, err = s3Client.CompleteMultipartUpload(ctx, completeInput1)
```

Comment:
check `err` returned from complete multipart as later it will be hard to understand why the list fail to work.